libs/template/templates/lakeflow-pipelines/databricks_template_schema.vnext.json

```diff
@@ -0,0 +1,78 @@
+    "properties": {
+        "project_name": {
+            "type": "string",
+            "default": "my_project",
+            "description": "\nUnique name for this project",
+            "order": 1,
+            "pattern": "^[A-Za-z0-9_]+$",
+            "pattern_match_failure_message": "Name must consist of letters, numbers, and underscores."
+        },
+        "project_name_short": {
+            "//": "This is a derived property based on project_name (it replaces my_project with sample and strips _project|_app|_service)",
+            "skip_prompt_if": {},
+            "type": "string",
+            "default": "{{if eq .project_name \"my_project\"}}sample{{else}}{{with (regexp \"^(my_)?(.*)(_project|_app|_service)?$\").FindStringSubmatch .project_name}}{{index . 2}}{{else}}{{.project_name}}{{end}}{{end}}",
+            "description": "Short name for the project",
+            "order": 2
+        },
+        "include_job": {
+            "type": "string",
+            "default": "yes",
+            "enum": ["yes", "no"],
+            "description": "Include a Lakeflow job that runs a notebook",
+            "order": 3
+        },
+        "include_pipeline": {
+            "type": "string",
+            "default": "yes",
+            "enum": ["yes", "no"],
+            "description": "Include a Lakeflow ETL pipeline",
+            "order": 4
+        },
+        "include_python": {
+            "type": "string",
+            "default": "yes",
+            "enum": ["yes", "no"],
+            "description": "Include a sample Python package that is built to a wheel file",
+            "order": 5
+        },
+        "serverless": {
+            "type": "string",
+            "default": "yes",
+            "enum": ["yes", "no"],
+            "description": "Use serverless compute",
+            "order": 6
+        },
+        "default_catalog": {
+            "type": "string",
+            "default": "{{default_catalog}}",
+            "pattern": "^\\w*$",
+            "pattern_match_failure_message": "Invalid catalog name.",
+            "description": "Default catalog for any tables created by this project{{if eq (default_catalog) \"\"}} (leave blank when not using Unity Catalog){{end}}",
+            "order": 7
+        },
+        "personal_schemas": {
+            "type": "string",
+            "description": "Use a personal schema for each user working on this project\n(this is recommended, your personal schema will be '{{.default_catalog}}.{{short_name}}')",
+            "default": "yes",
+            "enum": [
+                "yes",
+                "no (advanced: I will customize the schema configuration later in databricks.yml)"
```

Comment:
why don't we ask for the schema here?

Reply:
In the scenario where users don't select the per-user schema then they need to decide what they want to do instead. Things get complicated:
* Some customers want one catalog per user
* Some customers want some kind of schema-mangling approach
* Some customers might be okay with just a single hardcoded schema for dev and a single hard-coded schema for prod, but this seem so limiting that it seems better to use `default` and ask customers to configure things themselves if they don't like this.
* Some customers may do yet other things.